clippy_lints/src/precedence.rs

```diff
@@ -86,11 +103,16 @@ impl EarlyLintPass for Precedence {
         }
 
         if let ExprKind::Unary(UnOp::Neg, ref rhs) = expr.kind {
-            if let ExprKind::MethodCall(_, ref args) = rhs.kind {
+            if let ExprKind::MethodCall(ref path_segment, ref args) = rhs.kind {
                 if let Some(slf) = args.first() {
                     if let ExprKind::Lit(ref lit) = slf.kind {
                         match lit.kind {
                             LitKind::Int(..) | LitKind::Float(..) => {
+                                for &odd_function in &ODD_FUNCTIONS_WHITELIST {
+                                    if odd_function == &*path_segment.ident.name.as_str() {
+                                        return;
+                                    }
+                                }
```

Comment:
We could use an iterator here for more idiomatic rust:

````rust
if ODD_FUNCTIONS_WHITELIST
    .iter()
    .any(|odd_function| **odd_function == *path_segment.ident.name.as_str())
{
    return;
};
````

Reply:
Integrated